liblangutil/SourceLocation.cpp

```diff
@@ -34,6 +34,7 @@ SourceLocation const parseSourceLocation(std::string const& _input, std::string
 
 	boost::algorithm::split(pos, _input, boost::is_any_of(":"));
 
+	// TODO What to do with sourceIndex?
```

Comment:
What do you mean? It is still good to validate it for well-formedness (i.e. `-1 .. maxIndex`)

Reply:
yes

Reply:
It's still weird that the index is ignored and name is used without cross-checking it with the index.